antiplag/views.py

```diff
@@ -36,18 +42,28 @@ def post(self, request):
         if is_file:
             files = request.FILES.getlist("files")
 
+            if len(files) > MAX_FILES_PER_REQUEST:
+                 return Response(
+                     {"error": "More than max allowed files per request"}, status=status.HTTP_400_BAD_REQUEST
+                 )
+
             if not files:
                 return Response(
                     {"error": "No files present"}, status=status.HTTP_400_BAD_REQUEST
                 )
 
             for file in files:
-                Document.objects.create(
-                    file=file,
-                    name=file.name,
-                    submission=submission,
-                    type=Document.DocumentType.FILE,
-                )
+                if file.size < MAX_FILE_SIZE:
+                    Document.objects.create(
+                        file=file,
+                        name=file.name,
+                        submission=submission,
+                        type=Document.DocumentType.FILE,
+                        )
+                else:
```

Comment:
This means, if theres one file in the queue, which is larger than the limit, it will send the error response, and all the files followed in the queue will be discarded - is it the correct solution?

Shouldn't we either refuse all the files, or save only those remaining in the request which were smaller than the limit?

That's the business question.

Reply:
Question discussed internally, and conclusion was not to implement as partial success is not an acceptable status.